franklin/builder/models.py

```diff
@@ -39,34 +154,32 @@ class Site(models.Model):
         (SUCCESS, 'Deploy Succeeded'),
         (FAILED, 'Deploy Failed')
     )
-
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    owner = models.CharField(max_length=100, default='')
-    owner_id = models.PositiveIntegerField(blank=True, null=True)
-    repo_name = models.CharField(max_length=100)
-    repo_name_id = models.PositiveIntegerField(blank=True, null=True)
-    git_hash = models.CharField(max_length=40)
-    url = models.CharField(max_length=100, default='', db_index=True)
-    path = models.CharField(max_length=100)
-    oauth_token = models.CharField(max_length=255, default='')
+    
+    site = models.ForeignKey(Site, related_name='environments')
+    name = models.CharField(max_length=100, default='')
+    description = models.TextField(max_length=20480, default='', blank=True)
+    deploy_type = models.CharField(
+        max_length=3, choices=DEPLOY_CHOICES, default=BRANCH)
+    branch = models.CharField(max_length=100, default='master')
+    tag_regex = models.CharField(max_length=100, blank=True)
+    url = models.CharField(
+        max_length=100, default='', db_index=True, blank=True)
+    current_deploy = models.ForeignKey(
+        Build, related_name='deployments', null=True, blank=True)
+    past_builds = models.ManyToManyField(
+        Build, related_name='environments', blank=True)
```

Comment:
`current_deploy` and `past_builds` will be important for those advanced features abrahams wants for rolling back deployments and the manual `promote` deployment option. You'll notice that promote is listed as a `deploy_type`. This will get used after we build the admin panel for users. If they promote a deployment from...say....`staging` to `production`, we'll just assign the `current_deploy` FK of prod to point to the same build that staging's FK already points to. :boom: